### ISSUE TITLE
Unused Variables and hyperparameters in get_trimmed_set defense/weak.py

### ISSUE THREAD

**Issue by skoffas** [charlesjin/compatibility_clustering#1]
Hi,

I am working on this defense which I found very interesting to reproduce and the code well-written. However, I have the following questions about `get_trimmed_set` in `defense/weak.py`:

1) Some of the function arguments are not used in the function's main body (`predicted`, `indices_to_keep`, `labels`). Additionally, the variable `alpha` set in line 197 is not used. Is this on purpose or something was missed?

2) Why in line 202 `class_thresh` is `thresh / 8`? Is `8` related to the number of classes? Shall it be different if another dataset is used or this is the default hyperparameter based on your experiments?

**Comment by charlesjin** [comment]
Hey, thanks for checking out this work!

I'm going to do my best to answer your questions, but honestly it's been a while so let me know if you still have questions:

1. I believe `predicted`, `indices_to_keep`, and `labels` are all leftover arguments from debugging statements I removed when cleaning up the code, so you should be safe to remove them. Similarly, I believe that `alpha` is also a bit of leftover code that was originally used in place of the current `thresh` calculation. What probably happened is I opted for a simpler thresholding schedule, but forgot to delete the old code.
2.  `8` is a hyperparameter that is dataset-independent. In appendix C: "When trimming the training set, we also additionally include the top p/8 samples per class to prevent the network from collapsing to a trivial solution."

Hope this helps! Feel free to create a PR removing the extraneous code.

**Comment by skoffas** [comment]
Thank you so much for the reply. Now everything is clear and I have created a PR with these changes.
